src/SOS/SOS.Extensions/DebuggerServices.cs

```diff
@@ -37,22 +37,22 @@ internal DebuggerServices(IntPtr punk, HostType hostType)
             _hostType = hostType;
         }
 
-        public HResult GetOperatingSystem(out DebuggerServices.OperatingSystem operatingSystem)
```

Comment:
You don' t need to change the wrapper function's (GetOperationgSystem, etc.) return to int just the pinvoke delegate (VTable.GetOperatingSystem). That saves all kinds of unnecessary changes and still gives us the advantages of HResult but still fixes the pinvoke problem on Linux x86. 

Could you change all these back to HResult?

Reply:
ok, done.

Reply:
I can't do so for DataTargetServices.cs, etc.. because the return value type of the delegate should be the same as the method's one. If I do something like this: 

        private HResult GetMachineType(
        ...
        [UnmanagedFunctionPointer(CallingConvention.Winapi)]
        private delegate int GetMachineTypeDelegate(
            [In] IntPtr self,
            [Out] out IMAGE_FILE_MACHINE machineType);

I immediately get a build error message:
/home/oleg/work/fork/diagnostics/src/SOS/SOS.Hosting/DataTargetWrapper.cs(77,58): error CS0407: 'HResult DataTargetWrapper.GetMachineType(IntPtr, out IMAGE_FILE_MACHINE)' has the wrong return type [/home/oleg/work/fork/diagnostics/src/SOS/SOS.Hosting/SOS.Hosting.csproj]

Do I misunderstood something?

Reply:
Sorry, I was wrong. The COM callable wrappers (that inherit COMCallableIUnknown)  like DataTargetWrapper, HostWrapper.cs, etc. need both the delegate and function to return int like you are doing in this PR.

It is callable COM wrappers (that inherit CallableCOMWrapper) like DebuggerSetvices.cs, RemoteMemoryService.cs, etc. that the function can return HResult but the function pointer returns int (also like you have done). The only function pointers returning HResult that you missed are in ICorDebugController.cs, ICorDebug.cs and ICLRDebugging.cs in the src\tests\DbgShim.UnitTests tests.